src/components/shapes/calc_autorange.js

```diff
@@ -77,15 +73,35 @@ function calcPaddingOptions(lineWidth, sizeMode, v0, v1, path, isYAxis) {
     }
 }
 
-function shapeBounds(ax, v0, v1, path, paramsToUse) {
-    var convertVal = (ax.type === 'category' || ax.type === 'multicategory') ? ax.r2c : ax.d2c;
+function shapeBounds(ax, shape, paramsToUse) {
+    var dim = ax._id.charAt(0) === 'x' ? 'x' : 'y';
+    var isCategory = ax.type === 'category' || ax.type === 'multicategory';
+    var v0;
+    var v1;
+    var shiftStart = 0;
+    var shiftEnd = 0;
+
+    var convertVal = isCategory ? ax.r2c : ax.d2c;
```

Comment:
This could be simplified now that we only support category axis types.

Reply:
Do you mean, moving the shiftStart/shiftEnd into the convertVal function?
I didn't do that because in the case of sizemode `pixel` I think I need to apply convertVal but not the shift?
Not sure though, is this correct?

Reply:
Is there a case where the value of `isCategory` is *false*?

Reply:
yes, this function is called for shapes on all axes, so it requires a category check

Reply:
Thanks for the note.
Resolving.